### PR TITLE
Update jupyter_server version to 2.7.0 to match Determined AI

### PR DESCRIPTION
**Pull request by jerryharrow**
# Summary

Determined's task environments have jupyter_server==2.7.0, but jupyterlab-pachyderm is locked to  jupyter_server>=1.6,<2 which prevents it from being installed into a Determined task environment.  Change the locked version to jupyter_server==2.7.0 to match Determined as of 0.24.0.   jupyter_server recommends that the version be pinned
to avoid plug-in breakage.

The integration tests use an unauthenticated interface between the tests and the dev_server.   Add an jupyter_server identity provider that disables authentication to enable this to continue working with jupyter_server >= 2.0.

# Testing

Verified the plug-in mount capability with this configuration.
Are there other specific areas that we need to verify?

`docker run -it -p 8888:8888 --cap-add SYS_ADMIN --device /dev/fuse   pachyderm/notebooks-user:dev`

image

### CODE REVIEW

**Comment by msteffen** [comment]
For posterity, the auth changes in this PR are discussed in the comments on #9342  (and in [this comment](https://github.com/pachyderm/pachyderm/pull/9342#issuecomment-1729318561) in particular)

**Comment by jerryharrow** [comment]
> @jerryharrow thank you for figuring out the auth issue! @bbonenfant is planning to review this, since he has the most expertise.
> 
> I have one lingering question, which is: it looks like we created `dev_server.py` because there's nothing to call `_load_jupyter_server_extension` in our integration test suite, and we need a way to start a server. Why would updating our `jupyter_server` dependency affect that code, though? I thought maybe doing so updated tornado too, but at least on my machine, tornado is at 6.3.3 before and after the change.
> 
> @albscui I think you're the original author of `dev_server.py`—any chance you know?

The `dev-server` uses the `setup_handlers(web_app)` from the jupyterlab_pachyderm/handlers.py which all inherit from 
`jupyter_server.base.handlers.APIHandler`  and thus inherits the [jupyter_server.auth.IdentityProvider](https://jupyter-server.readthedocs.io/en/stable/operators/security.html#jupyter_server.auth.IdentityProvide).   The calls to the dev-server which come from test-integrations.py test are directly using `requests` without any authorization.

**Comment by jerryharrow** [comment]
In particular here is the failure from the dev-server startup log showing the stack running into /upyter_server/auth/identity.py from tornado.web.

```

Uncaught exception GET /pachyderm/v2/config (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/pachyderm/v2/config', version='HTTP/1.1', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/circleci/project/jupyter-extension/venv/lib/python3.9/site-packages/tornado/web.py", line 1765, in _execute
    result = await result  # type: ignore
  File "/home/circleci/project/jupyter-extension/venv/lib/python3.9/site-packages/jupyter_server/base/handlers.py", line 723, in prepare
    await super().prepare()
  File "/home/circleci/project/jupyter-extension/venv/lib/python3.9/site-packages/jupyter_server/base/handlers.py", line 614, in prepare
    _user = await _user
  File "/home/circleci/project/jupyter-extension/venv/lib/python3.9/site-packages/jupyter_server/auth/identity.py", line 240, in _get_user
    _cookie_user = self.get_user_cookie(handler)
  File "/home/circleci/project/jupyter-extension/venv/lib/python3.9/site-packages/jupyter_server/auth/identity.py", line 398, in get_user_cookie
    _user_cookie = handler.get_secure_cookie(
  File "/home/circleci/project/jupyter-extension/venv/lib/python3.9/site-packages/tornado/web.py", line 836, in get_signed_cookie
    self.require_setting("cookie_secret", "secure cookies")
  File "/home/circleci/project/jupyter-extension/venv/lib/python3.9/site-packages/tornado/web.py", line 1669, in require_setting
    raise Exception(
Exception: You must define the 'cookie_secret' setting in your application to use secure cookies
```

File: jupyter-extension/setup.cfg [hunk 1]
```diff
@@ -19,7 +19,7 @@ classifiers =
 packages = find:
 install_requires =
   pyyaml>=6.0
-  jupyter_server>=1.6,<2
+  jupyter_server==2.7.0
```
Comment (bbonenfant) [inline jupyter-extension/setup.cfg:22]: We should be able to loosen this pinning to `>=2.7.0,<2.8` to allow patches/bug-fixes to bubble up.

File: jupyter-extension/setup.cfg [hunk 2]
```diff
@@ -34,4 +34,5 @@ dev=
   pytest
   pytest-asyncio
   pytest-tornasync
+  pytest-jupyter
```
Comment (bbonenfant) [inline jupyter-extension/setup.cfg:37]: Why was this added? I don't see it used in your changes

**Review by bbonenfant — COMMENTED** [review]
I was able to run this locally. I don't have any issues with these changes being added, but I'd personally prefer if we merged them from a branch where CI is runnning. Is there any way you could push these changes to the branch @msteffen created and we can merge that in?

**Comment by msteffen** [comment]
> if we merged them from a branch where CI is runnning

I think CI is running in this branch, no? Under “successful checks"—aren’t those our usual tests?

**Review by bbonenfant — APPROVED** [review]
You're right @msteffen ... don't I look foolish

File: jupyter-extension/setup.cfg (hunk 2 above)
Comment (jerryharrow) [reply jupyter-extension/setup.cfg:37]: @msteffen  added that one in his review, and I copied it.   I can try dropping it.

**Comment by jerryharrow** [comment]
> You're right @msteffen ... don't I look foolish

The original one was on a fork (but now I'm part of the "in crowd") :-)

File: jupyter-extension/setup.cfg (hunk 2 above)
Comment (msteffen) [reply jupyter-extension/setup.cfg:37]: Oh, yeah. When I tried creating a testing branch from the original pr, CI failed with an error about that as a missing dependency, so I added it 🤷. We should remove it if it’s not needed.

File: jupyter-extension/setup.cfg (hunk 1 above)
Comment (jerryharrow) [reply jupyter-extension/setup.cfg:22]: Can do.   Determined is fixed to the single 2.7.0, but that doesn't mean Pachyderm has to be.
The jupyter_server recommendation was to pin to a specific version to avoid incompatibilities.   
Certainly patch versions are likely to be OK, but their example was to use and exact version (see below).

https://jupyter-server.readthedocs.io/en/latest/developers/dependency.html

> If your project depends directly on Jupyter Server, be sure to watch Jupyter Server’s ChangeLog and pin your project to a version that works for your application. Major releases represent possible backwards-compatibility breaking API changes or features.
```
To pin your jupyter_server install to a specific version:

> pip install jupyter_server==1.0.0

```

Comment (jerryharrow) [reply jupyter-extension/setup.cfg]: Updated as suggested.
